### PR TITLE
feat: simplify its

### PR DESCRIPTION
**Pull request by Foivos**
[AXE-2490](https://axelarnetwork.atlassian.net/browse/AXE-2490)

[AXE-2490]: https://axelarnetwork.atlassian.net/browse/AXE-2490?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: contracts/interchain-token-service/InterchainTokenService.sol [hunk 1]
```diff
@@ -227,182 +216,99 @@ contract InterchainTokenService is
...
-    ) external payable whenNotPaused {
-        bytes32 tokenId = customTokenId(msg.sender, salt);
-        _deployStandardizedToken(tokenId, distributor, name, symbol, decimals, mintAmount, msg.sender);
-        address tokenAddress_ = standardizedTokenAddress(tokenId);
-        _deployTokenManager(tokenId, TokenManagerType.MINT_BURN, abi.encode(msg.sender.toBytes(), tokenAddress_));
+        emit CustomTokenIdClaimed(tokenId_, deployer_, salt);
+        if (bytes(destinationChain).length == 0) {
+            _deployTokenManager(tokenId_, tokenManagerType, params);
+        } else {
+            _deployRemoteTokenManager(tokenId_, destinationChain, gasValue, tokenManagerType, params);
```
Comment (milapsheth) [inline contracts/interchain-token-service/InterchainTokenService.sol:276]: This should check that token manager was already deployed on the current chain

File: contracts/interchain-token-service/InterchainTokenService.sol [hunk 2]
```diff
@@ -335,74 +270,45 @@ contract InterchainTokenService is
...
-            destinationChain,
-            gasValue
-        );
+
+        if (bytes(destinationChain).length == 0) {
+            address tokenAddress_;
+            tokenAddress_ = _deployInterchainToken(tokenId_, distributor, name, symbol, decimals);
+            _deployTokenManager(tokenId_, TokenManagerType.MINT_BURN, abi.encode(operator, tokenAddress_));
+        } else {
+            _deployRemoteInterchainToken(tokenId_, name, symbol, decimals, distributor, operator, destinationChain, gasValue);
```
Comment (milapsheth) [inline contracts/interchain-token-service/InterchainTokenService.sol:310]: Same, check if the token was deployed locally first?

File: contracts/interchain-token-service/InterchainTokenService.sol [hunk 3]
```diff
@@ -335,74 +270,45 @@ contract InterchainTokenService is
...
-            mintTo,
-            mintAmount,
-            operator_,
-            destinationChain,
-            gasValue
-        );
+
+        if (bytes(destinationChain).length == 0) {
+            address tokenAddress_;
+            tokenAddress_ = _deployInterchainToken(tokenId_, distributor, name, symbol, decimals);
```
Comment (milapsheth) [inline contracts/interchain-token-service/InterchainTokenService.sol:307]: Should we fail early if token manager was already deployed for this salt?

File: contracts/interchain-token-service/InterchainTokenService.sol (hunk 1 above)
Comment (Foivos) [reply contracts/interchain-token-service/InterchainTokenService.sol:276]: _deployTokenManager fails if we try to deploy again. We can check in _deployTokenManager also, but that would introduce a bit more gas for better error messages when trying to deploy duplicates.

File: contracts/interchain-token-service/InterchainTokenService.sol (hunk 2 above)
Comment (Foivos) [reply contracts/interchain-token-service/InterchainTokenService.sol:310]: see above

File: contracts/interchain-token-service/InterchainTokenService.sol (hunk 3 above)
Comment (Foivos) [reply contracts/interchain-token-service/InterchainTokenService.sol:307]: we could but what is the point, really?

**Review by milapsheth — APPROVED** [review]

**Review by deanamiel — APPROVED** [review]

**Pull request merged**